logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetMessageAssetUseCase.kt

```diff
@@ -72,7 +74,10 @@ internal class GetMessageAssetUseCaseImpl(
             CompletableDeferred(MessageAssetResult.Failure(it))
         }, { message ->
             val assetDownloadStatus = (message.content as MessageContent.Asset).value.downloadStatus
+            val assetUploadStatus = (message.content as MessageContent.Asset).value.uploadStatus
             val wasDownloaded: Boolean = assetDownloadStatus == SAVED_INTERNALLY || assetDownloadStatus == SAVED_EXTERNALLY
+            // assets uploaded by other clients have upload status NOT_UPLOADED
+            val alreadyUploaded: Boolean = assetUploadStatus == NOT_UPLOADED || assetUploadStatus == UPLOADED
```

Comment:
hmmm, this might be a bit dangerous. How can we differentiate between an asset not uploaded from a different client and my current one? shouldn't we add an extra check here?

Reply:
As I described in the comment, for all assets from proto we always set `Message.UploadStatus.NOT_UPLOADED` and when the current client wants to send an asset, the first status is `UPLOAD_IN_PROGRESS` so it looks like we can differentiate:
- NOT_UPLOADED - state set only for assets received from backend which means they are already on the backend, sometimes they may be missing remote data (if it's a preview asset) so in that case I added `&& content.value.shouldBeDisplayed` to be sure, but even without that it would still work exactly the same way it used to work before this PR
- UPLOAD_IN_PROGRESS, UPLOADED, FAILED_UPLOAD - states set only for assets sent by the current client